mahiru/rest/schemas.yaml

```diff
@@ -422,3 +422,59 @@ components:
             oneOf:
               - "$ref": "#/components/schemas/Party"
               - "$ref": "#/components/schemas/Site"
+
+    WireGuardEndpoint:
+      type: object
+      required:
+        - address
+        - port
+        - key
+      properties:
+        address:
+          description: IP address of the endpoint to connect to
+          type: string
+        port:
+          description: Port of the endpoint to connect to
+          type: integer
+        key:
+          description: WireGuard key of the endpoint to connect to
+          type: string
+
+    WireGuardConnectionRequest:
+      type: object
+      required:
+        - net
+        - endpoint
+      properties:
+        net:
+          description: VPN network id to use for the connection
+          type: integer
+        endpoint:
+          description: WireGuard endpoint for the client side
+          type: object
+          schema:
+            "$ref": "#/components/schemas/WireGuardEndpoint"
+
+    ConnectionRequest:
+      oneOf:
+        - $ref: '#/components/schemas/WireGuardConnectionRequest'
```

Comment:
Cool, didn't know you could do something like this in `openapi`. I like it!

Reply:
Well, yes. Actually making it work with the Python library requires some acrobatics, and figuring out how to do that requires a dive into the source code because it's not really documented. See https://github.com/p1c2u/openapi-schema-validator/issues/5 :-)